crates/oxc_linter/src/rules/jest/prefer_jest_mocked.rs

```diff
@@ -0,0 +1,331 @@
+            None,
+            None,
+            None,
+        ),
+        (
+            "(<jest.Mock>foo).mockReturnValue(1);",
+            None,
+            None,
+            Some(PathBuf::from("/prefer-jest-mocked.ts")),
+        ),
+        ("(foo as jest.Mock).mockImplementation(1);", None, None, None),
+        ("(foo as unknown as jest.Mock).mockReturnValue(1);", None, None, None),
+        (
+            "(<jest.Mock>foo as unknown).mockReturnValue(1);",
+            None,
+            None,
+            Some(PathBuf::from("/prefer-jest-mocked.ts")),
+        ),
+        ("(Obj.foo as jest.Mock).mockReturnValue(1);", None, None, None),
+        ("([].foo as jest.Mock).mockReturnValue(1);", None, None, None),
+        ("(foo as jest.MockedFunction).mockReturnValue(1);", None, None, None),
+        ("(foo as jest.MockedFunction).mockImplementation(1);", None, None, None),
+        ("(foo as unknown as jest.MockedFunction).mockReturnValue(1);", None, None, None),
+        ("(Obj.foo as jest.MockedFunction).mockReturnValue(1);", None, None, None),
+        (
+            "(new Array(0).fill(null).foo as jest.MockedFunction).mockReturnValue(1);",
+            None,
+            None,
+            None,
+        ),
+        ("(jest.fn(() => foo) as jest.MockedFunction).mockReturnValue(1);", None, None, None),
+        (
+            "const mockedUseFocused = useFocused as jest.MockedFunction<typeof useFocused>;",
+            None,
+            None,
+            None,
+        ),
+        (
+            "const filter = (MessageService.getMessage as jest.Mock).mock.calls[0][0];",
+            None,
+            None,
+            None,
+        ),
+        (
+            "
+                class A {}
+                (foo as jest.MockedClass<A>)
+            ",
+            None,
+            None,
+            None,
+        ),
+        ("(foo as jest.MockedObject<{method: () => void}>)", None, None, None),
+        ("(Obj['foo'] as jest.MockedFunction).mockReturnValue(1);", None, None, None),
+        (
+            "
+                (
+                new Array(100)
+                    .fill(undefined)
+                    .map(x => x.value)
+                    .filter(v => !!v).myProperty as jest.MockedFunction<{
+                    method: () => void;
+                }>
+                ).mockReturnValue(1);
+            ",
+            None,
+            None,
+            None,
+        ),
+    ];
+
+    let fix = vec![
+        ("(foo as jest.Mock).mockReturnValue(1);", "(jest.mocked(foo)).mockReturnValue(1);"),
+        (
+            "(foo as unknown as string as unknown as jest.Mock).mockReturnValue(1);",
+            "(jest.mocked(foo)).mockReturnValue(1);",
+        ),
+        (
+            "(foo as unknown as jest.Mock as unknown as jest.Mock).mockReturnValue(1);",
+            "(jest.mocked(foo)).mockReturnValue(1);",
+        ),
+        // Note: couldn't fix
```

Comment:
Curious about why these cases couldn't be fixed

Reply:
Because Parser supports JSX by default, thus it will occur an error when parses `<`. And currently fixer couldn't set the file name to `.ts` like `expect` or `fail`.